src/vite/plugins/virtual-routes.ts

```diff
@@ -48,7 +48,7 @@ export function virtualRoutes(): PluginOption {
 
           let pagePath = path.replace(replacer, '').replace(/\.(.*)/, '')
           if (pagePath.endsWith('index'))
-            pagePath = pagePath.replace('index', '').replace(/\/$/, '')
```

Comment:
you don't want to replace all `index` in the pagePath into an empty string, it will break on pagePath such as `indexer-ref/index`